core/bin/external_node/src/config/mod.rs

```diff
@@ -244,6 +244,26 @@ pub struct OptionalENConfig {
     pub contracts_diamond_proxy_addr: Option<Address>,
 }
 
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct ApiComponentConfig {
+    /// Address of the tree API used by this EN in case it does not have a
+    /// local tree component running and in this case needs to send requests
+    /// to some external tree API.
+    pub tree_api_url: Option<String>,
+    pub tree_api_port: Option<u16>,
+    /// Proxy requests to `eth_syncing` to this node.
+    /// Used in case of running an API component of the EN together with a separate
+    /// instance of EN running `core` component. In that case it is the `core` component
+    /// that may do the actual syncing, not the API and so API has to forward these
+    /// requests to `core`.
+    pub ask_syncing_status_from: Option<String>,
```

Comment:
Won't this force such core instances to also have to run their own api ? If yes, were you considering using db for fetching current sync state or is something preventing us from using it that way?

Reply:
good idea

Reply:
Yeah idk, currently this is an in-memory piece of data. So if we put it into DB then the core node may be down but the DB will still have "syncing" in it and API will return it

Reply:
like @slowli 's approach the only thing is that this will result in more requests to the Main Node, and what's worse it'll scale linearly with the number of API servers.

On the other hand, the requests themselves are extremely cheap so maybe it's OK

Reply:
I don't have a strong opinion however with the current way:
1. one source of truth
2. a simple proxying of a request to it
3. APIs will need outbound connectivity, not only within the "distributed setup"

replicate `SyncState`:
1. many unsynced "sources of truth"
2. still doing some requests but somewhere outside of EN "distributed setup"

Reply:
Yep idk to me doing requests from API nodes to core nodes for this within an isolated distributed setup sounds more consistent then spamming O(n) requests to main node instead.

Reply:
The main node can be queried using a background task with a reasonably large interval between queries, so that rate-limiting is not a concern. Regarding rate-limiting, [I'm working](https://github.com/matter-labs/zksync-era/pull/1500) on applying it globally (within a process, of course), so that it would be even less of a concern in the near future. What I don't like about exposing the API server for the core is that it's not the user might expect, and that the entire API server is exposed (which might even have security repercussions) while we only use a single method.

Reply:
@slowli done